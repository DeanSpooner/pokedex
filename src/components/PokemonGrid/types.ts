export interface Props {
  region: string;
}
