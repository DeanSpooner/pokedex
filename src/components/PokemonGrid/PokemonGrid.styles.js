import styled from "styled-components";

export const GridWrapper = styled.div`
  flex-wrap: wrap;
  text-align: center;
`;

export const PokemonSearch = styled.input`
  border: solid #555 4px;
  border-radius: 16px;
  display: block;
  font-family: "Open Sans", sans-serif;
  font-size: 20px;
  margin: 24px auto;
  padding: 0 12px;
  height: 50px;
  width: 425px;
  text-align: center;
`;
