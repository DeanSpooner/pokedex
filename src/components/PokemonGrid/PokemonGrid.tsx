import * as React from "react";
import { useEffect, useState } from "react";
import { Props } from "./types";
import { GridWrapper } from "./PokemonGrid.styles";
import PokemonButton from "../PokemonButton/PokemonButton";
import axios from "axios";

export const PokemonGrid: React.FC<Props> = () => {
  const [pokemon, setPokemon] = useState([]);

  const fetchPokemon = async () => {
    const result = await axios("https://pokeapi.co/api/v2/pokemon?limit=151");

    await setPokemon(result.data.results);
    await console.log({ pokemon });
  };

  function capitalizeFirstLetter(string: string) {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }

  useEffect(() => {
    fetchPokemon();
  }, []);

  return (
    <GridWrapper>
      {pokemon.map((monster, index) => (
        <PokemonButton
          pokeName={capitalizeFirstLetter(monster?.name)}
          dexNumber={index + 1}
          spritePath={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
            index + 1
          }.png`}
        />
      ))}
    </GridWrapper>
  );
};

export default PokemonGrid;
