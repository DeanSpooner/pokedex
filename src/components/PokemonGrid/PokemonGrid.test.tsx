/** @jest-environment jsdom */

import React from "react";
import { render } from "@testing-library/react";
import PokemonGrid from "./PokemonGrid";

describe("PokemonGrid", () => {
  it("matches snapshot", () => {
    const { getByTestId } = render(<PokemonGrid region="kanto" />);
    expect(getByTestId("PokemonGrid")).toMatchSnapshot();
  });
});
