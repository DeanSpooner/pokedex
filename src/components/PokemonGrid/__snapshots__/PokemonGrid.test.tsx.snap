// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonGrid matches snapshot 1`] = `
<div
  class="PokemonGridstyles__GridWrapper-sc-1x2xoh6-0 fPNQOF"
  data-testid="PokemonGrid"
>
  <input
    class="PokemonGridstyles__PokemonSearch-sc-1x2xoh6-1 fAmQoN"
    placeholder="Search for a PokÃ©mon name or Dex number!"
    type="text"
  />
</div>
`;
